### PR TITLE
👾Prevent losing the game at the first click

### PR DESCRIPTION
**Pull request by javierbyte**
Currently you may lost at the first click. I solved that by checking before losing if it was your first click, if so, restart the game and excecute the click event on the same cell.

### CODE REVIEW

**Comment by muan** [comment]
Ah, I didn't think of restarting the game, nice solution! :sparkles: Though currently it looks a bit complicated. I think we can get away without saving `lastClick` and x/y on the elements. 

How about checking for number of elements with the `unmasked` class, and if that equals 1, just click the coordinates of that one `unmasked` cell. Then we can just click based on the classnames of the element. It'd be something like `cell x4 y7 unmasked`, so we'd ignore `unmasked`, and just find & click `.cell.x4.y7` directly after restarting.

And to restart, you should call `restart()` instead of `init()` directly. :)

:crocodile: :baby_chick:

**Comment by javierbyte** [comment]
Good! I've updated the pull request, what do you think now? :)

File: game.js [hunk 1]
```diff
@@ -51,6 +51,15 @@ Game.prototype.bindEvents = function () {
   Array.prototype.forEach.call(cells, function (target) {
     target.addEventListener('click', function () {
       if (!target.isMasked || target.isFlagged) return false
+      if (target.isBomb) {
+        var unmaskedCount = document.getElementsByClassName('cell unmasked').length
+        if(unmaskedCount === 0) {
```
Comment (muan) [inline game.js]: We can just merged this condition with `isBomb` at L54.

File: game.js [hunk 2]
```diff
@@ -51,6 +51,15 @@ Game.prototype.bindEvents = function () {
   Array.prototype.forEach.call(cells, function (target) {
     target.addEventListener('click', function () {
       if (!target.isMasked || target.isFlagged) return false
+      if (target.isBomb) {
+        var unmaskedCount = document.getElementsByClassName('cell unmasked').length
```
Comment (muan) [inline game.js]: Don't need `cell` here

File: game.js [hunk 3]
```diff
@@ -51,6 +51,15 @@ Game.prototype.bindEvents = function () {
...
+        if(unmaskedCount === 0) {
+          that.restart(that.twemoji)
+          var targetClasses = target.getAttribute('class').replace('unmasked', '')
+          document.getElementsByClassName(targetClasses)[0].click()
+          return;
```
Comment (muan) [inline game.js]: No semicolons. :grimacing:

File: game.js [hunk 4]
```diff
@@ -51,6 +51,15 @@ Game.prototype.bindEvents = function () {
   Array.prototype.forEach.call(cells, function (target) {
     target.addEventListener('click', function () {
       if (!target.isMasked || target.isFlagged) return false
+      if (target.isBomb) {
+        var unmaskedCount = document.getElementsByClassName('cell unmasked').length
+        if(unmaskedCount === 0) {
+          that.restart(that.twemoji)
+          var targetClasses = target.getAttribute('class').replace('unmasked', '')
```
Comment (muan) [inline game.js]: `.className` will do what `getAttribute('class')` does.

**Comment by javierbyte** [comment]
Done with the comments :)

**Comment by muan** [comment]
:raised_hands: :raised_hands: :raised_hands: :raised_hands: :raised_hands: :raised_hands: 
thank you so much this is great!!!!

**Pull request merged**